CMakeLists.txt

```diff
@@ -460,109 +470,114 @@ if(MSVC)
 endif()
 
 if(BASEARCH_X86_FOUND)
-    # Check whether compiler supports SSE2 instrinics
-    if(WITH_NATIVE_INSTRUCTIONS)
-        set(CMAKE_REQUIRED_FLAGS "${NATIVEFLAG}")
-    else()
-        set(CMAKE_REQUIRED_FLAGS "${SSE2FLAG}")
-    endif()
-    check_c_source_compile_or_run(
-        "#include <immintrin.h>
-        int main(void)
-        {
-            __m128i zero = _mm_setzero_si128();
-            (void)zero;
-            return 0;
-        }"
-        HAVE_SSE2_INTRIN
-    )
-    set(CMAKE_REQUIRED_FLAGS)
-
-    # Check whether compiler supports SSE4 CRC inline asm
-    if(WITH_NATIVE_INSTRUCTIONS)
-        set(CMAKE_REQUIRED_FLAGS "${NATIVEFLAG}")
-    else()
-        set(CMAKE_REQUIRED_FLAGS "${SSE4FLAG}")
-    endif()
-    check_c_source_compile_or_run(
-        "int main(void)
-        {
-            unsigned val = 0, h = 0;
-        #if defined(_MSC_VER)
-            { __asm mov edx, h __asm mov eax, val __asm crc32 eax, edx __asm mov val, eax }
-        #else
-            __asm__ __volatile__ ( \"crc32 %1,%0\" : \"+r\" (h) : \"r\" (val) );
-        #endif
-            return (int) h;
-        }"
-        HAVE_SSE42CRC_INLINE_ASM
-    )
-    # Check whether compiler supports SSE4 CRC instrinics
-    check_c_source_compile_or_run(
-        "#include <immintrin.h>
-        int main(void)
-        {
-            unsigned crc = 0;
-            char c = 'c';
-        #if defined(_MSC_VER)
-            crc = _mm_crc32_u32(crc, c);
-        #else
-            crc = __builtin_ia32_crc32qi(crc, c);
-        #endif
-            (void)crc;
-            return 0;
-        }"
-        HAVE_SSE42CRC_INTRIN
-    )
-    set(CMAKE_REQUIRED_FLAGS)
+    if(WITH_SSE2)
+        # Check whether compiler supports SSE2 instrinics
+        if(WITH_NATIVE_INSTRUCTIONS)
+            set(CMAKE_REQUIRED_FLAGS "${NATIVEFLAG}")
+        else()
+            set(CMAKE_REQUIRED_FLAGS "${SSE2FLAG}")
+        endif()
+        check_c_source_compile_or_run(
+            "#include <immintrin.h>
+            int main(void)
+            {
+                __m128i zero = _mm_setzero_si128();
+                (void)zero;
+                return 0;
+            }"
+            HAVE_SSE2_INTRIN
+        )
+        set(CMAKE_REQUIRED_FLAGS)
 
-    # Check whether compiler supports PCLMULQDQ intrinics
-    if(WITH_NATIVE_INSTRUCTIONS)
-        set(CMAKE_REQUIRED_FLAGS "${NATIVEFLAG}")
-    else()
-        set(CMAKE_REQUIRED_FLAGS "${PCLMULFLAG}")
+        # FORCE_SSE2 option will only be shown if HAVE_SSE2_INTRIN is true
+        if("${ARCH}" MATCHES "i[3-6]86")
+            cmake_dependent_option(FORCE_SSE2 "Always assume CPU is SSE2 capable" OFF "HAVE_SSE2_INTRIN" OFF)
+        endif()
```

Comment:
This doesn't actually disable SSE2 as the value of HAVE_SSE2_INTRIN is cached. It must be explicitly cleared.

Reply:
Fixed.